presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -2210,6 +2210,15 @@ public void testJoinAggregations()
     }
 
     @Test
+    public void testProjectionPushThroughJoin()
+            throws Exception
+    {
+        assertQuery("SELECT discount - tax, orderstatus FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey");
+        assertQuery("SELECT discount - tax, totalprice * 2 FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey");
+        assertQuery("SELECT orderstatus, totalprice * (discount - tax) FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey");
```

Comment:
Can you add tests with nested joins/projections?

Also, add some tests to TestLogicalPlanner to ensure the rewrite works as expected (rather than just verifying the query produces the expected results).